src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -11,6 +13,19 @@ public AsyncRequestProcessor(Executor executor) {
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        Map<String, UserData> userDataMap = new ConcurrentHashMap<>();
+
+        if (userDataMap.containsKey(userId)) {
+            try {
+                Thread.sleep(500);
+            } catch (InterruptedException e) {
+                throw new RuntimeException();
+            }
+            return CompletableFuture.supplyAsync(() -> userDataMap.get(userId));
```

Comment:
The use of `Thread.sleep(500)` inside an asynchronous method is questionable. If the intent is to simulate a delay, use `CompletableFuture.delayedExecutor` to introduce the delay in the returned `CompletableFuture` instead of blocking the current thread.